morpheus/lib/morpheus_web/controllers/product_controller.ex

```diff
@@ -0,0 +1,48 @@
+defmodule MorpheusWeb.ProductController do
+  use MorpheusWeb, :controller
+
+  alias Morpheus.Products
+  alias Morpheus.Products.Product
+
+  action_fallback(MorpheusWeb.FallbackController)
+
+  def index(conn, _params) do
+    products = Products.list_products() |> Enum.reject(fn product -> product.is_deleted end)
+    render(conn, :index, products: products)
+  end
+
+  def create(conn, %{"product" => product_params}) do
+    with {:ok, %Product{} = product} <- Products.create_product(product_params) do
+      conn
+      |> put_status(:created)
+      |> put_resp_header("location", ~p"/api/products/#{product}")
+      |> render(:show, product: product)
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    product = Products.get_product!(id)
```

Comment:
Tutaj taka jedna kwestia. Używasz funkcji z wykrzyknikiem na końcu.

Różnica jest taka
- `Products.get_product` - zwróci ci albo product albo `nil` jeżeli nie znaleziony
-  `Products.get_product!` - zwróci ci albo product, a jeżeli nie znaleziony to wywoła raise

Tutaj lepiej skorzystać z `Products.get_product` tej wersji.

Reply:
Przy wykorzystaniu nowej funkcji robisz tutaj.

```
case Product.get_product(id) do
    {:ok, product} ->
        render(conn, :show, product: product)
    {:error, :deleted} ->
        conn |> put_status(:not_found) |> json("Item has been deleted")
    _ ->
        conn |> put_status(:not_found)
end
```